coap/CMakeLists.txt

```diff
@@ -1,38 +1,44 @@
-if(NOT CONFIG_LWIP_IPV6 AND NOT CMAKE_BUILD_EARLY_EXPANSION)
-    message(STATUS "IPV6 support is disabled so the coap component will not be built")
-    # note: the component is still included in the build so it can become visible again in config
-    # without needing to re-run CMake. However no source or header files are built.
-    idf_component_register()
-    return()
-endif()
-
 set(include_dirs port/include port/include libcoap/include)
 
 set(srcs
-    "port/src/block.c"
     "libcoap/src/coap_address.c"
     "libcoap/src/coap_asn1.c"
     "libcoap/src/coap_async.c"
+    "libcoap/src/coap_block.c"
     "libcoap/src/coap_cache.c"
     "libcoap/src/coap_debug.c"
+    "libcoap/src/coap_dtls.c"
+    "libcoap/src/coap_encode.c"
     "libcoap/src/coap_event.c"
     "libcoap/src/coap_hashkey.c"
     "libcoap/src/coap_io.c"
+    "libcoap/src/coap_layers.c"
+    "libcoap/src/coap_mbedtls.c"
+    "libcoap/src/coap_mem.c"
+    "libcoap/src/coap_net.c"
+    "libcoap/src/coap_netif.c"
     "libcoap/src/coap_notls.c"
     "libcoap/src/coap_option.c"
+    "libcoap/src/coap_oscore.c"
+    "libcoap/src/coap_pdu.c"
     "libcoap/src/coap_prng.c"
+    "libcoap/src/coap_resource.c"
     "libcoap/src/coap_session.c"
+    "libcoap/src/coap_str.c"
     "libcoap/src/coap_subscribe.c"
     "libcoap/src/coap_tcp.c"
     "libcoap/src/coap_time.c"
-    "libcoap/src/encode.c"
-    "libcoap/src/mem.c"
-    "libcoap/src/net.c"
-    "libcoap/src/pdu.c"
-    "libcoap/src/resource.c"
-    "libcoap/src/str.c"
-    "libcoap/src/uri.c"
-    "libcoap/src/coap_mbedtls.c")
+    "libcoap/src/coap_uri.c"
+    "libcoap/src/coap_ws.c")
```

Comment:
Should this be added under `CONFIG_COAP_WEBSOCKETS`?

Reply:
No.  `coap_ws.c` continues to provide public API functions that handle `CONFIG_COAP_WEBSOCKETS `is not defined and return doing the appropriate nothing.